### PR TITLE
allow to relaunch all stuck jobs

### PR DESCRIPTION
**Pull request by eryshev**
- new route /api/executions/relaunch?jobs=
- new method in Executor `def relaunch` that  is invoked by new route
- code rearrangement in Executor

### CODE REVIEW

File: core/src/main/scala/com/criteo/cuttle/Executor.scala [hunk 1]
```diff
@@ -602,10 +605,16 @@ class Executor[S <: Scheduling] private[cuttle] (val platforms: Seq[ExecutionPla
...
+    atomic { implicit txn =>
+      execution.startTime = Some(Instant.now())
+      runningState += (execution -> promise.future)
+    }
+
   private def unsafeDoRun(execution: Execution[S], promise: Promise[Completed]): Unit =
     promise.completeWith(
-      execution.job
-        .run(execution)
+      execution
```
Comment (dufrannea) [inline core/src/main/scala/com/criteo/cuttle/Executor.scala]: Same as above, not sure why it had to change

File: core/src/main/scala/com/criteo/cuttle/Executor.scala [hunk 2]
```diff
@@ -620,6 +629,7 @@ class Executor[S <: Scheduling] private[cuttle] (val platforms: Seq[ExecutionPla
             execution.streams.error(stacktrace.toString)
             atomic {
               implicit tx =>
+                // retain jobs in a recent failures if a last failure happened in [now - retryStrategy.retryWindow, now]
```
Comment (dufrannea) [inline core/src/main/scala/com/criteo/cuttle/Executor.scala]: May I suggest removing both 'a'
// retain jobs in **a** recent failures if **a** last failure happened in [now - retryStrategy.retryWindow, now]

File: core/src/main/scala/com/criteo/cuttle/Executor.scala [hunk 3]
```diff
@@ -262,6 +263,12 @@ case class Execution[S <: Scheduling](
           isWaiting.set(false)
       }
     }
+
+  /**
+    * Run this execution on its job.
+    */
+  def run(): Future[Completed] =
```
[the remark this answers is not in the document]
Comment (eryshev) [reply core/src/main/scala/com/criteo/cuttle/Executor.scala]: It's  just more convenient to call this instead
```
execution.job.run(execution)
```
But I agree with you that it should be private.

File: core/src/main/scala/com/criteo/cuttle/Executor.scala (hunk 2 above)
Comment (eryshev) [reply core/src/main/scala/com/criteo/cuttle/Executor.scala]: You may and I will correct it ty!

**Pull request merged**